network/common/src/main/java/org/apache/spark/network/client/TransportResponseHandler.java

```diff
@@ -50,6 +55,8 @@
 
   private final Map<Long, RpcResponseCallback> outstandingRpcs;
 
+  private final Queue<StreamCallback> streamCallbacks;
```

Comment:
are you sure this can just be a queue?  I would have expected a `Map<StreamId, StreamCallback>`.  You don't know the order you'll get the stream responses back, right?

Reply:
Sorry to comment on this very old thread.

I get it that this can be a queue, but I don't get it why this can't be a `Map<StreamId, StreamCallback>`. Is it for performance concerns?

Reply:
ok so we can use a map here, but we pick queue as it's enough and simpler?

Reply:
It's simpler and it enforces the same semantics as the RPC contract (FIFO).